components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/util/JSONResponseWriter.java

```diff
@@ -115,14 +132,221 @@ private static JsonObject abstractResultToJSONObject(AbstractResult result) thro
             jsonResult.add(EntitlementEndpointConstants.ASSOCIATED_ADVICE, advices);
         }
 
+        // If includeInResponse=true, other attributes will be populated from here with the decision.
+        if (((Result) result).getAttributes() != null && !((Result) result).getAttributes().isEmpty()) {
+            Set<Attributes> attributes = ((Result) result).getAttributes();
+
+            for (Attributes attribute : attributes) {
+
+                switch (attribute.getCategory().toString()) {
+                    case EntitlementEndpointConstants.CATEGORY_ACTION_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_ACTION, getJsonObject(attribute));
+                        break;
+
+                    case EntitlementEndpointConstants.CATEGORY_RESOURCE_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_RESOURCE, getJsonObject(attribute));
+                        break;
+
+                    case EntitlementEndpointConstants.CATEGORY_ACCESS_SUBJECT_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_ACCESS_SUBJECT, getJsonObject(attribute));
+                        break;
+
+                    case EntitlementEndpointConstants.CATEGORY_ENVIRONMENT_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_ENVIRONMENT, getJsonObject(attribute));
+                        break;
+
+                    case EntitlementEndpointConstants.CATEGORY_RECIPIENT_SUBJECT_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_RECIPIENT_SUBJECT, getJsonObject(attribute));
+                        break;
+
+                    case EntitlementEndpointConstants.CATEGORY_INTERMEDIARY_SUBJECT_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_INTERMEDIARY_SUBJECT, getJsonObject(attribute));
+                        break;
+
+                    case EntitlementEndpointConstants.CATEGORY_CODEBASE_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_CODEBASE, getJsonObject(attribute));
+                        break;
+
+                    case EntitlementEndpointConstants.CATEGORY_REQUESTING_MACHINE_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_REQUESTING_MACHINE, getJsonObject(attribute));
+                        break;
+                }
+            }
+
+        }
+
         /**
-         * Todo: Category, PolicyIdentifierList
+         * Todo: PolicyIdentifierList
```

Comment:
what is this todo? can you provide bit more details(like a jira link) so that some other person can pick up and work on later

Reply:
and change it from doc comment to a block comment or single line comment